qiskit/transpiler/passes/layout/apply_layout.py

```diff
@@ -59,7 +59,7 @@ def run(self, dag):
         for creg in dag.cregs.values():
             new_dag.add_creg(creg)
         for node in dag.topological_op_nodes():
-            if node.type == "op":
+            if isinstance(node, OpNode):
```

Comment:
Some of the existing `node.type == "op"` checks are no longer needed, since `dag.topological_op_nodes()` will only return `OpNode`s. Doesn't have to be done as part of this PR, but it would be good at some point to identify and remove these.

Reply:
I think there are only a handful of these left that are sure OpNode's, so I can probably take care of that here.

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/6567/commits/205ac2606600c88cf47ebc652623e1ca1fa68035